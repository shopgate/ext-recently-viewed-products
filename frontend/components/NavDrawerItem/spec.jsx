import React from 'react';
import { mount } from 'enzyme';
import Item from './index';

// eslint-disable-next-line require-jsdoc, react/prop-types
const MockedItem = props => (<div>{ props.children }</div>);

let mockedIsiOSTheme = false;
jest.mock('../../helpers/isiOSTheme', () => () => mockedIsiOSTheme);

describe('NavDrawer', () => {
  it('should render the Menu item', () => {
    const component = mount(<Item Item={MockedItem} handleClose={() => {}} />);
    expect(component).toMatchSnapshot();
    expect(component.html()).not.toBeNull();
  });

  it('should render nothing for ios', () => {
    mockedIsiOSTheme = true;
    const component = mount(<Item Item={MockedItem} handleClose={() => {}} />);
    expect(component.html()).toBeNull();
  });
});
