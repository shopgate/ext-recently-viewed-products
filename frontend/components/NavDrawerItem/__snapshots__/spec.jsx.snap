// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavDrawer should render the Menu item 1`] = `
<NavDrawerItem
  Item={[Function]}
  handleClose={[Function]}
>
  <MockedItem
    close={[Function]}
    icon={[Function]}
    link="/recently-viewed"
    title="recently_viewed_products.title"
  >
    <div>
      <Translate
        className=""
        params={Object {}}
        string="recently_viewed_products.title"
      >
        <span
          className=""
        >
          recently_viewed_products.title
        </span>
      </Translate>
    </div>
  </MockedItem>
</NavDrawerItem>
`;
